src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.c

```diff
@@ -64,92 +138,990 @@ const ActorInit En_Grasshopper_InitVars = {
+    Math_ApproachZeroF(&this->damagedVelocity.z, 1.0f, 2.0f);
+    if ((this->action != EN_GRASSHOPPER_ACTION_FALL) && (this->type != EN_GRASSHOPPER_TYPE_WOODFALL)) {
+        Actor_UpdateBgCheckInfo(play, &this->actor, 35.0f, 60.0f, 60.0f, 0x1D);
+    }
+
+    this->actor.shape.rot.z = this->actor.world.rot.z;
+    if (this->type == EN_GRASSHOPPER_TYPE_GROWS_WHEN_SPAWNED) {
+        f32 targetScale = this->type * 0.01f;
+
+        if (targetScale >= 0.018f) {
+            targetScale = 0.018f;
+        }
+
+        Math_ApproachF(&this->dragonflyScale, targetScale, 0.1f, 0.01f);
+    }
+
+    if ((this->action != EN_GRASSHOPPER_ACTION_FALL) && (this->action != EN_GRASSHOPPER_ACTION_APPROACH_PLAYER)) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 5, 0x3E8, 5);
+    }
+
+    EnGrasshopper_UpdateEffects(this, play);
+    if ((this->action != EN_GRASSHOPPER_ACTION_DEAD) && (this->action != EN_GRASSHOPPER_ACTION_FALL)) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+        if ((this->action == EN_GRASSHOPPER_ACTION_APPROACH_PLAYER) || (this->action == EN_GRASSHOPPER_ACTION_ATTACK)) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+        }
+    }
+}
+
+void EnGrasshopper_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnGrasshopper* this = THIS;
+    Vec3f sEffectOffsetFromTailTop = { 500.0f, 0.0f, 0.0f };
+    Vec3f sZeroVec3f = { 0.0f, 0.0f, 0.0f };
+
+    Matrix_Push();
+
+    if ((this->type != EN_GRASSHOPPER_TYPE_WOODFALL_TEMPLE_FINAL_ROOM) &&
+        (this->type != EN_GRASSHOPPER_TYPE_WOODFALL)) {
+        if ((s8)(sLimbIndexToShadowBodyPartsIndex[limbIndex] >= 0)) {
+            Matrix_MultVec3f(&sZeroVec3f, &this->shadowBodyPartsPos[sLimbIndexToShadowBodyPartsIndex[limbIndex]]);
+        }
+    }
+
+    if (limbIndex == DRAGONFLY_LIMB_TAIL_TIP) {
+        Matrix_Translate(0.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_MultVec3f(&gZeroVec3f, &this->tailTipPos);
+        Matrix_MultVec3f(&sEffectOffsetFromTailTop, &this->effectBasePos);
+    }
+
+    Collider_UpdateSpheres(limbIndex, &this->collider);
+    if ((limbIndex == DRAGONFLY_LIMB_ROOT) || (limbIndex == DRAGONFLY_LIMB_TAIL_SEGMENT_2) ||
+        (limbIndex == DRAGONFLY_LIMB_TAIL_TIP) || (limbIndex == DRAGONFLY_LIMB_BACK_LEFT_UPPER_LEG) ||
+        (limbIndex == DRAGONFLY_LIMB_BACK_RIGHT_UPPER_LEG) || (limbIndex == DRAGONFLY_LIMB_FRONT_LEFT_UPPER_LEG) ||
+        (limbIndex == DRAGONFLY_LIMB_FRONT_LEFT_FOOT) || (limbIndex == DRAGONFLY_LIMB_FRONT_RIGHT_UPPER_LEG) ||
+        (limbIndex == DRAGONFLY_LIMB_HEAD) ||
+        //! @bug: These do not check for valid limbs. Limb index 0x18 is DRAGONFLY_LIMB_MAX, so it (and any index
+        //! larger than it) is not tied to an actual limb.
+        (limbIndex == 0x18) || (limbIndex == 0x19) ||
```

Comment:
Oh what a mess, this may be connected to the weird size of the shadow array too. Also they're indices, so should be in dec.